test/unit/forms/FormValidationSpec.scala

```diff
@@ -124,14 +124,50 @@ class FormValidationSpec extends UnitSpec {
       val res  = nameDobForm.bind(data)
       res.errors should not be empty
     }
-    "fail when a date of birth is missing" in {
+    "fail when date of birth - day and month missing" in {
       val data = formData.updated("date-of-birth.day", "").updated("date-of-birth.month", "")
       val res  = nameDobForm.bind(data)
       res.errors shouldBe Seq(
-        FormError("date-of-birth.day", Seq("dob.error.empty-date")),
-        FormError("date-of-birth.month", "")
+        FormError("date-of-birth.day", List("date-of-birth.day-date-of-birth.month.empty"), List()),
+        FormError("date-of-birth.month", List(""), List())
       )
     }
+    "fail when date of birth - day and year missing" in {
```

Comment:
Added tests for the new scenarios, kept in other tests for each form to maintain code coverage